### ISSUE TITLE
bug: getPlatforms() returning incorrect platforms

### ISSUE THREAD

**Issue by ebk46** [ionic-team/ionic#19942]
# Bug Report

**Ionic version:**
[x] **4.11.5 (React)**

**Current behavior:**
Using getPlatforms() on 2 Windows full-size laptops yielded:
Laptop 1 (Chrome): ["phablet", "mobile", "mobileweb"]
Laptop 1 (Firefox): ["mobile", "mobileweb"]
Laptop 2 (Chrome): ["mobile", "mobileweb"]

Laptop 1 is a Dell Precision M3800
Laptop 2 is a standard issue corporate Windows machine at a Fortune 100 company (not sure the make/model)

**Expected behavior:**
I would expect a full size computer (laptop) to have "desktop" as one of its platforms and would expect NOT to see "mobile", "mobileweb", or "phablet".

**Steps to reproduce:**
I made a StackBlitz that simply shows the output of getPlatforms(), so using it on various machines should expose the issue (below).

**Related code:**
https://stackblitz.com/edit/ionic-v4-11-5-getplatforms

**Other information:**
I do NOT see this issue on MacOS, only Windows (only tested Windows 10 as well).

**Ionic info:**
```
Ionic:

   Ionic CLI       : 5.4.6 (/Users/evan/.config/yarn/global/node_modules/ionic)
   Ionic Framework : @ionic/react 4.11.5

Capacitor:

   Capacitor CLI   : 1.3.0
   @capacitor/core : 1.3.0

Utility:

   cordova-res : 0.6.0 (update available: 0.8.1)
   native-run  : not installed

System:

   NodeJS : v12.10.0 (/usr/local/Cellar/node/12.10.0/bin/node)
   npm    : 6.11.3
   OS     : macOS High Sierra
```

**Comment by warntjo** [comment]
I also see getPlatforms() returning wrong values. In my case on a mobile (Motorola G6 Plus) it returns: ["android", "cordova", "desktop", "hybrid"].

I've tested in a new ionic angular app created with "ionic start myApp tabs".

On other mobile phones the same app returns "mobile" instead of "desktop".

**Comment by ebk46** [comment]
This is really impacting my deployment. I need to handle Capacitor plugins that don't have web implementations by using them only for mobile deployments (e.g. Keyboard) but I can't reliably do that. There are other reasons this is impactful as well. Any updates or at least temporary workarounds?

**Comment by mxdmedia** [comment]
I am seeing the same issue as @ebk46, with a Dell XPS All-in-One.  `platform.getPlatforms()` returns `['mobile', 'mobileweb']`.    One thing that seems to be common between my issue and @ebk46's experience is both machines have a touch screen (and they both are dells).  I wonder if it has something to do with how the machine reports its touchscreen abilities to windows?

**Comment by peterpeterparker** [comment]
I just faced this one today to, `platform.is('mobile')` was returning `true` on my Sony with Windows  OS with a touchscreen. A friend of mine noticed something similar on Windows too so I understood it as `mobile` being true generally speaking on Windows.

Note: it was with Ionic v5, I didn't tried with v4

My workaround: adding a test for Windows

```
export function isWindows(): boolean {
  if (!window || !navigator) {
    return false;
  }

  const a: string = navigator.userAgent || navigator.vendor || (window as any).opera;

  return /windows/i.test(a);
}
```

**Comment by mxdmedia** [comment]
Another update.  One of my clients just purchased an Asus Zenbook Duo, and it too is exhibiting this behavior.  So this isn't just limited to Dell machines.  Also of note, I took a deeper look at what some laptops are reporting- a 13" Macbook Pro is being recognized as ["desktop", "phablet"].  The desktop seems correct.. the "phablet"?  Not so much.

**Comment by tylerc** [comment]
I'm chiming in here to report the same issue with touch screen 2-in-1 laptops - they get detected as "mobile" rather than "desktop", which caused a decent amount of confusion during testing.

The root cause seems to be some faulty assumptions in https://github.com/ionic-team/ionic/blob/master/core/src/utils/platform.ts

"desktop" is defined as "not mobile" and "mobile" is defined as anything that matches the media query `any-pointer: coarse` ( https://developer.mozilla.org/en-US/docs/Web/CSS/@media/any-pointer )

It seems that a decent alternative might be `any-pointer: fine` for desktop, even `pointer: fine`.

**Comment by royalrex** [comment]
I can confirm that browsers on windows computers that have a touch screen connected will give a false positive to be "mobile".

On my setup, I have 2 non-touch monitors connected to my laptop, the laptop monitor is a touchscreen, but not enabled together with the other monitors. It still give me "touch" capabilities. 

Other websites as well give me popups about not being touch optimized, so it's not only Ionic that could do a better check.

**Comment by monkeytronics** [comment]
Yeah. Same as above. I managed to get some sensible behaviour with :

```
    if ( !this.platform.is('cordova') ) {
      this.bigScreen = true;
    }
```
I got the same baffling problem with mobile & mobileweb, and it's a real shame. It would be very nice to render the app in a mobile like view when using the web portal on mobile.

**Comment by JensUweB** [comment]
Can confirm, getPlatforms() returns "android" on an virtual apple device (xcode).
An alternative for everyone that uses capacitor: [https://capacitorjs.com/docs/apis/device](https://capacitorjs.com/docs/apis/device)

**Comment by SmallhillCZ** [comment]
The problem is here:
 - https://github.com/ionic-team/ionic-framework/blob/b0d53ca73619585671d8cf4dc24e47f826495a0a/core/src/utils/platform.ts#L91-L95
 
All in one comuputers report both `(any-pointer:coarse)` and `(any-pointer:fine)`, because they can be operated by finger or mouse. Ionic the chooses only by presence of `(any-pointer:coarse)`.

**Comment by stankovicmilutin** [comment]
For me, using Dell XPS 15 7590 with TouchScreen I get from 
`platform.platforms()` 
 `["mobile", "mobileweb"]`

Any workaround?

**Comment by NadavPeled1998** [comment]
I use [ua-parser-js](https://www.npmjs.com/package/ua-parser-js) library instead of isPlatform/getPlatform to solve this issue.
